test/jdk/com/sun/jdi/JdwpOnThrowTest.java

```diff
@@ -0,0 +1,161 @@
+            VirtualMachine vm = null;
+            try {
+                vm = attach("localhost", "" + port);
+                EventQueue queue = vm.eventQueue();
+                log("Waiting for exception event");
+                long start = System.currentTimeMillis();
+                while (start + TIMEOUT > System.currentTimeMillis()) {
+                    EventSet eventSet = queue.remove(TIMEOUT);
+                    EventIterator eventIterator = eventSet.eventIterator();
+                    while(eventIterator.hasNext() && start + TIMEOUT > System.currentTimeMillis()) {
+                        Event event = eventIterator.next();
+                        if (event instanceof ExceptionEvent ex) {
+                            if (ex.exception() == null) {
+                                throw new RuntimeException("Exception is null");
+                            }
+                            if (ex.exception().type() == null) {
+                                throw new RuntimeException("Exception type is null");
+                            }
+                            if (ex.exception().referenceType() == null) {
+                                throw new RuntimeException("Exception reference type is null");
+                            }
+                            if (ex.catchLocation() == null) {
+                                throw new RuntimeException("Exception catch location is null");
+                            }
+                            if (!ex.location().equals(ex.thread().frame(0).location())) {
+                                throw new RuntimeException(
+                                    String.format("Throw location %s and location of first frame %s are not equal",
+                                                  ex.location(), ex.thread().frame(0).location()));
+                            }
+                            if (!ex.exception().type().name().equals("Ex")) {
+                                throw new RuntimeException("Exception has wrong type: " + ex.exception().type().name());
+                            }
+                            log("Received exception event: " + event);
+                            vm.dispose();
+                            return;
+                        }
+                        log("Received event: " + event);
+                    }
+                }
+                throw new RuntimeException("ERROR: failed to receive exception event");
+            } catch (IOException ex) {
+                throw new RuntimeException("ERROR: failed to attach", ex);
+            }
+        }
+    }
+
+    private static int findFreePort() {
+        try (ServerSocket socket = new ServerSocket(0)) {
+            return socket.getLocalPort();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static String ATTACH_CONNECTOR = "com.sun.jdi.SocketAttach";
```

Comment:
move the `private static` fields up to the other place where `private static long TIMEOUT = 10000;` is declared to have them together.